### ISSUE TITLE
SIGALRM doesn't work on Windows

### ISSUE THREAD

**Issue by justaddcoffee** [yaseminbridges/phenotype2phenopacket#24]
our colleague Elena Casiraghi is seeing this error when running `p2p` on Windows, presumably b/c SIGALRM isn't a thing on Windows:

```
C:\Users\ele\AppData\Local\pypoetry\Cache\virtualenvs\ga-UqBDSXhM-py3.11\Lib\site-packages\linkml_runtime\linkml_model\model\schema\types does not look like a va
lid URI, trying to serialize this will break.
Downloading https://purl.obolibrary.org/obo/hp/hpoa/phenotype.hpoa to data\phenotype.hpoa
C:\Users\ele\AppData\Local\pypoetry\Cache\virtualenvs\ga-UqBDSXhM-py3.11\Lib\site-packages\pronto\ontology.py:206: UnicodeWarning: unsound encoding, assuming ISO
-8859-1 (73% confidence)
  return cls(
Creating synthetic patient phenopackets:   0%|                                                                                         | 0/8170 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "C:\Users\ele\AppData\Local\pypoetry\Cache\virtualenvs\ga-UqBDSXhM-py3.11\Scripts\\ga", line 6, in <module>
    sys.exit(main())
             ^^^^^^
  File "C:\Users\ele\AppData\Local\pypoetry\Cache\virtualenvs\ga-UqBDSXhM-py3.11\Lib\site-packages\click\core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ele\AppData\Local\pypoetry\Cache\virtualenvs\ga-UqBDSXhM-py3.11\Lib\site-packages\click\core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "C:\Users\ele\AppData\Local\pypoetry\Cache\virtualenvs\ga-UqBDSXhM-py3.11\Lib\site-packages\click\core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ele\AppData\Local\pypoetry\Cache\virtualenvs\ga-UqBDSXhM-py3.11\Lib\site-packages\click\core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ele\AppData\Local\pypoetry\Cache\virtualenvs\ga-UqBDSXhM-py3.11\Lib\site-packages\click\core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\DATI\Anacleto\N3C_projects\new_GA_main\ga_pyphetools_cohorts\src\ga\cli_make_synthetic_data.py", line 84, in make_synthetic_data_command
    create_synthetic_patient_phenopacket(
  File "C:\Users\ele\AppData\Local\pypoetry\Cache\virtualenvs\ga-UqBDSXhM-py3.11\Lib\site-packages\phenotype2phenopacket\create\create.py", line 41, in create_sy
nthetic_patient_phenopacket
    patient_terms = synthetic_patient_generator.patient_term_annotation_set()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ele\AppData\Local\pypoetry\Cache\virtualenvs\ga-UqBDSXhM-py3.11\Lib\site-packages\phenotype2phenopacket\utils\phenopacket_utils.py", line 557, i
n patient_term_annotation_set
    patient_terms = self.get_patient_terms()
                    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ele\AppData\Local\pypoetry\Cache\virtualenvs\ga-UqBDSXhM-py3.11\Lib\site-packages\phenotype2phenopacket\utils\phenopacket_utils.py", line 383, i
n get_patient_terms
    return self.filter_phenotype_entries(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ele\AppData\Local\pypoetry\Cache\virtualenvs\ga-UqBDSXhM-py3.11\Lib\site-packages\phenotype2phenopacket\utils\phenopacket_utils.py", line 354, i
n filter_phenotype_entries
    signal.signal(signal.SIGALRM, handler)
                  ^^^^^^^^^^^^^^
AttributeError: module 'signal' has no attribute 'SIGALRM'. Did you mean: 'SIGABRT'?
```

One possible solution might be to use `SIGABRT` [here](https://github.com/yaseminbridges/phenotype2phenopacket/blob/4ab4e47732851cce3fa8c62f1b8abea59fddeb6c/src/phenotype2phenopacket/utils/phenopacket_utils.py#L355) instead of `SIGALRM`? Not sure, I don't have a Windows machine to check:

```
        time_limit = 15
        signal.signal(signal.SIGALRM, handler)  # <<<<<<<<<
        signal.alarm(time_limit)
        try:
            while len(self.filtered_df) < max_number:
                for phenotype_entry in frequency_df.rows(named=True):
                    if len(self.filtered_df) >= max_number:
                        break
                    self.check_frequency(phenotype_entry)
            return pl.from_dicts(self.filtered_df)
        except TimeoutError:
            if len(self.filtered_df) == 0:
                return frequency_df.sample(n=max_number)
            else:
                return pl.from_dicts(self.filtered_df)
        finally:
            signal.alarm(0)
```

**Comment by yaseminbridges** [comment]
Currently trying to fix in #25 - but I have another issue with one of the packages I use for the HPO that's causing a time out error

**Comment by yaseminbridges** [comment]
I've gone with your code but just replaced:

```python
if thread.is_alive():
            thread._stop()  # Forcefully stops the thread
            print("Timed out!")
```

with:

```python
thread.daemon = True
stop_event = threading.Event()
if thread.is_alive():
            stop_event.set()  # Forcefully stops the thread
            print("Timed out!")
```

As I got an assertion error with the `_stop()` method.

Yes, sometimes it gets stuck in almost an infinite loop, where some of the terms have a very low frequency this loop:

```python
                    for phenotype_entry in frequency_df.rows(named=True):
                        if len(self.filtered_df) >= max_number:
                            break
                        self.check_frequency(phenotype_entry)
```

will just repeat as the generated fake patient frequency won't meet the threshold for the frequency for the HPO term and it's unable to fulfil the number of terms specified.

**Comment by yaseminbridges** [comment]
Should be fixed now in #25

**Comment by justaddcoffee** [comment]
Thanks @yaseminbridges!
